File: ProcessSim.Core.CSharp/UnitConverter.cs
using System;

namespace ProcessSim.Core.CSharp
{
    public static class UnitConverter
    {
        public static double MgPerMlToGPerL(double value)
        {
            if (value < 0) throw new ArgumentOutOfRangeException(nameof(value), "La valeur ne peut pas être négative.");
            return value;
        }

        public static double GPerLToMgPerMl(double value)
        {
            if (value < 0) throw new ArgumentOutOfRangeException(nameof(value), "La valeur ne peut pas être négative.");
            return value;
        }

        public static double MlToL(double value)
        {
            if (value < 0) throw new ArgumentOutOfRangeException(nameof(value), "La valeur ne peut pas être négative.");
            return value / 1000.0;
        }

        public static double LToMl(double value)
        {
            if (value < 0) throw new ArgumentOutOfRangeException(nameof(value), "La valeur ne peut pas être négative.");
            return value * 1000.0;
        }

        public static double CelsiusToKelvin(double value) => value + 273.15;

        public static double KelvinToCelsius(double value) => value - 273.15;
    }
}

File: ProcessSim.Core/UnitConverter.vb
Option Strict On
Option Explicit On

Public NotInheritable Class UnitConverter
    Private Sub New()
    End Sub

    Public Shared Function MgPerMlToGPerL(value As Double) As Double
        If value < 0 Then Throw New ArgumentOutOfRangeException(NameOf(value))
        Return value
    End Function

    Public Shared Function GPerLToMgPerMl(value As Double) As Double
        If value < 0 Then Throw New ArgumentOutOfRangeException(NameOf(value))
        Return value
    End Function

    Public Shared Function MlToL(value As Double) As Double
        If value < 0 Then Throw New ArgumentOutOfRangeException(NameOf(value))
        Return value / 1000.0
    End Function

    Public Shared Function LToMl(value As Double) As Double
        If value < 0 Then Throw New ArgumentOutOfRangeException(NameOf(value))
        Return value * 1000.0
    End Function

    Public Shared Function CelsiusToKelvin(value As Double) As Double
        Return value + 273.15
    End Function

    Public Shared Function KelvinToCelsius(value As Double) As Double
        Return value - 273.15
    End Function
End Class
